src/utils/prim-lexing.fs

```diff
@@ -2,10 +2,105 @@
 
 #nowarn "47" // recursive initialization of LexBuffer
 
+namespace Microsoft.FSharp.Compiler.Text
+
+open System
+open System.IO
+
+type ISourceText =
+
+    abstract Item : int -> char with get
+
+    abstract GetLineString : lineIndex: int -> string
+
+    abstract GetLineCount : unit -> int
+
+    abstract GetLastCharacterPosition : unit -> int * int
+
+    abstract GetSubTextString : start: int * length: int -> string
+
+    abstract SubTextEquals : target: string * startIndex: int -> bool
+
+    abstract Length : int
+
+    abstract ContentEquals : sourceText: ISourceText -> bool
+
+    abstract CopyTo : sourceIndex: int * destination: char [] * destinationIndex: int * count: int -> unit  
+
+[<Sealed>]
+type StringText(str: string) =
+
+    let getLines (str: string) =
+        use reader = new StringReader(str)
+        [|
+        let line = ref (reader.ReadLine())
```

Comment:
Why a ref cell instead of mutable value? Do you explicitly want to allocate on the heap? Otherwise this is a bit clearer:

```fsharp
let mutable line = reader.ReadLine()
while not (isNull line) do
    yield line
    line <- reader.ReadLine()
```

Reply:
This is a copy paste from somewhere else in F#. I didn't think to improve it.